### PR TITLE
AdvLoggerPkg: Switch to using AsciiStrnLenS

### LINKED ISSUE #292: [Feature]: Switch AdvancedLogger to use Safe Strings.
### Feature Overview

BaseLib contains both AsciiStrLen and AsciiStrnLenS. Currently advacned logger is just using AsciiStrLen, but switching ot the safe string version does not require any additional changes.

### Solution Overview

Switch to using AsciiStrnLenS, with appropriate maximum string length parameters. 

### Alternatives Considered

_No response_

### Urgency

Low

### Are you going to implement the feature request?

I will implement the feature

### Do you need maintainer feedback?

No maintainer feedback needed

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by apop5**
## Description
A compounds assert was found in a platform. An assert was triggered, and when attempting to generate the assert messages, print lib triggered an assert as well. The system was found to have reached the PcdMaximumAsciiStringLength referenced in AsciiStrLen.

To combat this, the AsciiStrLen calls in AdvancedLogger are being switched to use the safe version, with the associated max lengths being the stack buffers that are used in the functions. 

This change allowed assert messages to be displayed.

Fixes #292 

- [X] Impacts functionality?
  - **Functionality** - Does the change ultimately impact how firmware functions?
  - Examples: Add a new library, publish a new PPI, update an algorithm, ...
- [ ] Impacts security?
  - **Security** - Does the change have a direct security impact on an application,
    flow, or firmware?
  - Examples: Crypto algorithm change, buffer overflow fix, parameter
    validation improvement, ...
- [ ] Breaking change?
  - **Breaking change** - Will anyone consuming this change experience a break
    in build or boot behavior?
  - Examples: Add a new library class, move a module to a different repo, call
    a function in a new library class in a pre-existing module, ...
- [ ] Includes tests?
  - **Tests** - Does the change include any explicit test code?
  - Examples: Unit tests, integration tests, robot tests, ...
- [ ] Includes documentation?
  - **Documentation** - Does the change contain explicit documentation additions
    outside direct code modifications (and comments)?
  - Examples: Update readme file, add feature readme file, link to documentation
    on an a separate Web page, ...

## How This Was Tested

Tested on a platform that was triggering asserts.
Tested in CI
## Integration Instructions

N/A - Changes should be

### CODE REVIEW

**Review by cfernald — APPROVED** [review]

File: AdvLoggerPkg/Library/AssertTelemetryLib/AssertLib.c [hunk 1]
```diff
@@ -86,7 +86,7 @@ DebugAssert (
   //
   // Send the print string to the Logging device device
   //
-  AdvancedLoggerWrite (DEBUG_ERROR, Buffer, AsciiStrLen (Buffer));
+  AdvancedLoggerWrite (DEBUG_ERROR, Buffer, AsciiStrnLenS (Buffer, sizeof (Buffer)Buffer));
```
Comment (makubacki) [inline AdvLoggerPkg/Library/AssertTelemetryLib/AssertLib.c]: Was the `Buffer` after `sizeof()` meant to be there?

```suggestion
  AdvancedLoggerWrite (DEBUG_ERROR, Buffer, AsciiStrnLenS (Buffer, sizeof (Buffer)));

```

File: PcBdsPkg/Include/Library/MsBootPolicyLib.h [hunk 2]
```diff
@@ -15,7 +15,7 @@ SPDX-License-Identifier: BSD-2-Clause-Patent
  * application is at PcsBdsPkg/MsBootPolicy.
  *
  * Boot Policy Applications can be created to meet a platform's requirements. The PcBdsPkg
- * example aplication impelements USB, PXE and HDD boot sequences, in addition to a default
+ * example application impelements USB, PXE and HDD boot sequences, in addition to a default
```
Comment (os-d) [inline PcBdsPkg/Include/Library/MsBootPolicyLib.h]: SpellCheck is in AuditOnly mode for PcBdsPkg. I don't have a problem with this spell fix here, but also probably worthwhile to turn off AuditOnly mode and fix all the spelling errors in a separate patch. Up to you.

**Review by os-d — APPROVED** [review]
With Michael's comment, LGTM.

**Review by makubacki — APPROVED** [review]

File: AdvLoggerPkg/Library/AssertTelemetryLib/AssertLib.c (hunk 1 above)
Comment (apop5) [reply AdvLoggerPkg/Library/AssertTelemetryLib/AssertLib.c]: Thanks for catching it.

File: PcBdsPkg/Include/Library/MsBootPolicyLib.h (hunk 2 above)
Comment (apop5) [reply PcBdsPkg/Include/Library/MsBootPolicyLib.h]: Sounds good. Will revert for now, and worry about it in a separate PR.

**Pull request merged**